packages/flutter/lib/src/services/binary_messenger.dart

```diff
@@ -55,6 +55,17 @@ abstract class BinaryMessenger {
   /// This is intended for testing. Messages intercepted in this manner are not
   /// sent to platform plugins.
   void setMockMessageHandler(String channel, Future<ByteData> handler(ByteData message));
+
+  /// Handles all the stored platform messages stored inside of the channel buffers.
+  ///
+  /// This method is intended to be called by implementations of [setMessageHandler] once
+  /// it is valid to get calls to [handlePlatformMessage] for the given channel.
+  void drainChannelBuffers(String channel) {
+    while (!ui.channelBuffers.isEmpty(channel)) {
+      final ui.StoredMessage message = ui.channelBuffers.pop(channel);
```

Comment:
Increasing the dart:ui API surface seems fine. I guess the alternative would have been to create some sort of implied consequence from calling the setter of window.onPlatformMessage. The advantage with this one is that it's a bit extensible. 

If ever in the future, we decide on the receiving side to disregard all the queued message from the sender side and only take the last one, we can augment the channelBuffers object.